test/modules/aws-ecs-simplified-integration.ts

```diff
@@ -0,0 +1,210 @@
+    VALUES ('${appName}', ${desiredCount}, ${appPort}, '${cpuMem}', '${imageTag}', false);
+  `));
+
+  it('applies adds a new row', apply());
+
+  it('check row insertion', query(`
+    SELECT *
+    FROM ecs_simplified
+    WHERE app_name = '${appName}';
+  `, (res: any[]) => expect(res.length).toBe(1)));
+
+  it('uninstalls the ECS module', uninstall(modules));
+
+  it('installs the ECS module', install(modules));
+
+  it('check row was imported after uninstall/install', query(`
+    SELECT *
+    FROM ecs_simplified
+    WHERE app_name = '${appName}';
+  `, (res: any[]) => {
+    expect(res[0]['repository_uri'].includes(ecrRepositoryUri)).toBeTruthy();
+    expect(res[0]['load_balancer_dns']).toBeDefined();
+    return expect(res.length).toBe(1);
+  }));
+
+  it('updates a row', query(`
+    UPDATE ecs_simplified
+    SET load_balancer_dns = 'invalid'
+    WHERE app_name = '${appName}';
+  `));
+
+  it('applies row update and should restore the dns value', apply());
+
+  it('check row was restored', query(`
+    SELECT *
+    FROM ecs_simplified
+    WHERE app_name = '${appName}';
+  `, (res: any[]) => {
+    expect(res[0]['load_balancer_dns'] === 'invalid').toBeFalsy();
+    return expect(res.length).toBe(1);
+  }));
+
+  it('updates a row', query(`
+    UPDATE ecs_simplified
+    SET app_port = ${appPort + 1}
+    WHERE app_name = '${appName}';
+  `));
+
+  it('applies row update and should replace', apply());
+
+  it('check row was replaced', query(`
+    SELECT *
+    FROM ecs_simplified
+    WHERE app_name = '${appName}';
+  `, (res: any[]) => {
+    expect(res[0]['repository_uri'].includes(ecrRepositoryUri)).toBeTruthy();
+    expect(res[0]['app_port']).toBe(appPort + 1);
+    return expect(res.length).toBe(1);
+  }));
+
+  it('updates a row', query(`
+    UPDATE ecs_simplified
+    SET repository_uri = NULL
+    WHERE app_name = '${appName}';
+  `));
+
+  it('applies row update and should restore using the right repository uri', apply());
+
+  it('check row was updated and restored', query(`
+    SELECT *
+    FROM ecs_simplified
+    WHERE app_name = '${appName}';
+  `, (res: any[]) => {
+    expect(res[0]['repository_uri'].includes(ecrRepositoryUri)).toBeTruthy();
+    return expect(res.length).toBe(1);
+  }));
+
+  it('updates a row', query(`
+    UPDATE ecs_simplified
+    SET cpu_mem = '${updateCpuMem}', image_tag = '${updateImageTag}'
+    WHERE app_name = '${appName}';
+  `));
+
+  it('applies row update and should update service', apply());
+
+  it('check row was updated', query(`
+    SELECT *
+    FROM ecs_simplified
+    WHERE app_name = '${appName}';
+  `, (res: any[]) => {
+    expect(res[0]['cpu_mem']).toBe(updateCpuMem);
+    expect(res[0]['image_tag']).toBe(updateImageTag);
+    return expect(res.length).toBe(1);
+  }));
+
+  it('deletes the app', query(`
+    delete from ecs_simplified
+    where app_name = '${appName}';
+  `));
+
+  it('applies deletes the app', apply());
```

Comment:
Should there be a check that there are no records in the DB after this apply?

Reply:
yes!